ansible/roles/master/templates/haproxy.cfg.j2

```diff
@@ -0,0 +1,73 @@
+kip to content
+This repository
+Search
+Pull requests
+Issues
+Gist
+ @adamschaub
+ Watch 3
+  Star 4
+ Fork 5 contiv/cluster
+ Code  Issues 7  Pull requests 0  Wiki  Pulse  Graphs
+Branch: master Find file Copy pathcluster/vendor/ansible/roles/haproxy/templates/precise/haproxy.cfg
+6987ae6  on Jan 4
+@mapuri mapuri use the new contiv/ansible repo
+1 contributor
+RawBlameHistory     59 lines (53 sloc)  1.45 KB
+#
+# {{ ansible_managed }}
+#
+global
+    log /dev/log    local0
+    log /dev/log    local1 notice
+    chroot /var/lib/haproxy
+    user haproxy
+    group haproxy
+    daemon
+    stats socket /var/lib/haproxy/stats level admin
+
+defaults
+    log global
+    mode    http
+    option  httplog
+    option  dontlognull
+    contimeout 5000
+    clitimeout 50000
+    srvtimeout 50000
+    errorfile 400 /etc/haproxy/errors/400.http
+    errorfile 403 /etc/haproxy/errors/403.http
+    errorfile 408 /etc/haproxy/errors/408.http
+    errorfile 500 /etc/haproxy/errors/500.http
+    errorfile 502 /etc/haproxy/errors/502.http
+    errorfile 503 /etc/haproxy/errors/503.http
+    errorfile 504 /etc/haproxy/errors/504.http
+
+frontend http_frontend
+    bind *:80
+    mode http
+    option httpclose
+    option forwardfor
+    default_backend rgw
+
+frontend https_frontend
+    bind *:443 ssl crt /etc/ceph/radosgw-key-cert.pem
+    mode http
+    option httpclose
+    option forwardfor
+    reqadd X-Forwarded-Proto:\ https
+    default_backend rgw
+
+backend rgw
+    mode http
+    balance roundrobin
+    cookie RADOSGWLB insert indirect nocache
+{% for host in groups[masters] %}
+    server {{ hostvars[host].ansible_hostname }} {{ hostvars[host]['ansible_' + etcd_interface ].ipv4.address }}:80 check cookie {{ hostvars[host].ansible_hostname }}
```

Comment:
Yes, this was a messy commit that included haproxy.cfg before I had edited it. I'll be removing this and submitting a separate lb role.

As for interface name, using ansible_default_ipv4 breaks the vagrant environment, because it binds to virtualboxes NAT interface (@rutsky can keep me honest on this point). Should we further add `apiserver_interface`?